### PR TITLE
WorkspaceStarter: fix + cleanup error handling

### PR DESCRIPTION
**Pull request by geropl**
## Description
This PR does three things:
 1. does away with the feature flag `workspace_start_check_pending_first`
 2. fixes a bug that failed to map `ScmStartError` to `StartInstanceError`
 3. merges the two cascading try...catch block into a single one

To be more safe while rolling this out, we also keep the old code around, and have the feature flag `workspace_start_old` to switch back to it should be notice anything odd.

:information_source: The PR is best reviewed a) commit-per-commit and b) with "split view"+"whitespace hidden":
image

## Related Issue(s)

Fixes ENT-178

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - gpl-178-scm-error</li>
	<li><b>🔗 URL</b> - <a href="https://gpl-178-scm-error.preview.gitpod-dev.com/workspaces" target="_blank">gpl-178-scm-error.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - gpl-178-scm-error-gha.25431</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-gpl-178-scm-error%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/server/src/workspace/workspace-starter.ts [hunk 1]
```diff
@@ -683,6 +688,163 @@ export class WorkspaceStarter {
         }
     }
 
+    private async actuallyStartWorkspaceOld(
```
Comment (geropl) [inline components/server/src/workspace/workspace-starter.ts:691]: This refactoring honestly looks intimidating in the GitHub diff view :confused: 
At it's core it's really not, but hard to proof. I wonder if I should go with the less invasive version just to ease a review... :thinking: 

Let me know what you think!

File: components/server/src/workspace/workspace-starter.ts [hunk 2]
```diff
@@ -1966,8 +2115,8 @@ function resolveGitpodTasks(ws: Workspace, instance: WorkspaceInstance): TaskCon
     return tasks;
 }
 
-export async function isCheckPendingFirstEnabled(user: { id: string }): Promise<boolean> {
-    return getExperimentsClientForBackend().getValueAsync("workspace_start_check_pending_first", false, {
+export async function isUseOldWorkspaceStartEnabled(user: { id: string }): Promise<boolean> {
+    return getExperimentsClientForBackend().getValueAsync("workspace_start_old", false, {
```
Comment (svenefftinge) [inline components/server/src/workspace/workspace-starter.ts:2119]: would it be better to opt into the new behavior instead of opting into the old one?

Comment (geropl) [reply components/server/src/workspace/workspace-starter.ts:2119]: Good call. Will do :+1:

**Review by svenefftinge — APPROVED** [review]

**Comment by svenefftinge** [comment]
I did not test this, though

File: components/server/src/workspace/workspace-starter.ts (hunk 2 above)
Comment (geropl) [reply components/server/src/workspace/workspace-starter.ts:2119]: Done.

**Comment by geropl** [comment]
I did swap the ff enabled logic to opt-in, and tested some of the edge cases in the preview again: :heavy_check_mark:

**Pull request merged**